source/s3_auto_ranged_get.c

```diff
@@ -397,18 +396,24 @@ static int s_s3_auto_ranged_get_prepare_request(
     aws_s3_request_setup_send_data(request, message);
     aws_http_message_release(message);
 
+    /* Success! */
     AWS_LOGF_DEBUG(
         AWS_LS_S3_META_REQUEST,
         "id=%p: Created request %p for part %d",
         (void *)meta_request,
         (void *)request,
         request->part_number);
 
-    return AWS_OP_SUCCESS;
+    success = true;
 
-message_alloc_failed:
-
-    return AWS_OP_ERR;
+finish:;
+    struct aws_future *future = aws_future_new(meta_request->allocator, AWS_FUTURE_VALUELESS);
+    if (success) {
+        aws_future_set_valueless(future);
```

Comment:
do you need to explicitly set valueless on a future that was created valueless?

Reply:
yes, in this design, you got a FATAL_ASSERT if you called a getter or setter that didn't align with the enum passed to new()

but in the latest design it's type-safe and you can't compile the wrong getter/setter.